src/main/java/com/cleanroommc/multiblocked/jei/IJeiIngredientAdapter.java

```diff
@@ -0,0 +1,15 @@
+package com.cleanroommc.multiblocked.jei;
+
+import mezz.jei.api.recipe.IIngredientType;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * @author youyihj
+ */
+public interface IJeiIngredientAdapter<T, R> extends Function<T, Stream<R>> {
+    Class<T> getInternalIngredientType();
+
+    IIngredientType<R> getJeiIngredientType();
```

Comment:
Optional.Method(), make sure mbd can work without JEI installed, tho it desnt make sense.